### ISSUE TITLE
Question: how to select in mode='time' a minimum before midnight and a maximum after midnight

### ISSUE THREAD

**Issue by ThomasStubbe** [henninghall/react-native-date-picker#444]
**Describe the bug**
I want to configure a time-input going from 16:00 till 4:00 after midnight:
```
let minDate: Date = new Date();
minDate.setDate(minDate.getDate() -1) //Subtract one day (without this, the date input flashes completely jumping from min to max)
minDate.setHours(16, 0, 0);
let maxDate: Date = new Date();
maxDate.setHours(4, 0, 0);
<DatePicker style={styles.clockView}
                        mode='time'
                        minimumDate={minDate}
                        maximumDate={maxDate}/>
```
With this implementation I can select hours from 16:00h till 23:59. When selecting > 00:00, it will automatically jump to 16:00.
Date is not relevant for me. I only want to set a time-window for the time that a person goes to bed.
I'm probably missing something really basic.

**Expected behavior**
I can select a time between 16:00 and 04:00

**To Reproduce**
Add example code that reproduces the behavior. 
```javascript
export default class App extends Component {

  state = { 
    date: new Date() 
  }

  render () {
    let minDate: Date = new Date();
    minDate.setDate(minDate.getDate() -1);
    minDate.setHours(16, 0, 0);
    let maxDate: Date = new Date();
    maxDate.setHours(4, 0, 0);
    return (
      <DatePicker
        date={this.state.date}
        minimumDate={minDate}
        maximumDate={maxDate}
        onDateChange={date => this.setState({ date })}
      />
    );
  }
}
```

**Smartphone (please complete the following information):**
 - OS: Android
 - React Native version: 0.63.4
 - react-native-date-picker version: 4.1.1

**Comment by VolkerLieber** [comment]
You can probably solve your issue like this:
```
<DatePicker
        mode="time"
        date={this.state.date}
        onDateChange={(date) => {
// TODO: Check if the date is between 04:00 and 16:00 and correct it with date,setHours
this.setState({date: date})
)}
      />
```

**Comment by ThomasStubbe** [comment]
Thanks for the suggestion. Like you suggested, I implemented my own min-max logic like this right now:

```
  if (!isNil(minimumValue) && !isNil(maximumValue)) {
      if (value < minimumValue && value + 1 < maximumValue) {
        value.setDate(value.getDate() + 1);
      }
      if (value > maximumValue && value - 1 > minimumValue) {
        value.setDate(value.getDate() + 1);
      }
    }

    if (!isNil(minimumValue) && value < minimumValue) {
      value = minimumValue;
    }
    if (!isNil(maximumValue) && value > maximumValue) {
      value = maximumValue;
    }
    //...
```

This works. However, I noticed a second issue with this library: this solution will only correct the time a first time (e.g. choosing 15h when the max is 14h => it will go back to 14h), however choosing a time > 14h again will not update the component, because the state actually did not change (it stays 14h). A `this.forceUpdate()` also doesn't work. 
So basically the component does not know that it is displaying a different value than what the state is saying. Any ideas how to solve this?

**Comment by henninghall** [comment]
Hi! I've investigated this issue a bit. I understand your problem. I've tried the iOS picker and it seems to have the same or a similar issue. Since the iOS picker is built on the underlaying native component UIDatePicker I don't really see a way to solve this properly. Even if it would be possible to fix on Android the issue would still persist on iOS.

Unfortunately, I think this has to be considered a limitation of this library, at least until it is supported in [UIDatePicker](https://developer.apple.com/documentation/uikit/uidatepicker)

This workaround would be a custom validation logic, something similar to what you've written.
